pkg/recommendation/capacity.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright 2020 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recommendation
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"mayadata.io/cstorpoolauto/types"
+	bdutil "mayadata.io/cstorpoolauto/util/blockdevice"
+)
+
+// capacityRecommendationRequest contains Block device list
+// and raid group config. This is the request schema to get
+// min and max pool capacity for a given raid type.
+type capacityRecommendationRequest struct {
```

Comment:
Capacity* ?

Reply:
I am forcing the consumer of this package to create the request using `NewCapacityRequest()` as all the validation is in this function. Here the input is only `RaidGroupConfig` If we want to make it public raid group config will be the only input.
capacityRecommendationRequest struct is a wrapper over actual public request schema.

For now, we have rest call In rest call request will be only `RaidGroupConfig` and caller will provide block device list and raidGroupConfig to this package to create a request and get a response.
I think we should not expose this wrapper structure.

Reply:
This comment is based on the fact that request should be considered as an input argument.
The constructor or method or function that takes in this request should/may want to do request validation before proceeding.